syntax = "proto3";

package compn;

service CompressorNode {
  rpc SendData(SensorBatch) returns (Ack);
  rpc SendCompressData(SensorBatch) returns (Ack);
}

// Leitura do sensor.
message SensorReading {
  int32 value = 1;
  int64 timestamp = 2;
}

// Conjunto de leituras
message SensorBatch {
  int32 sensor_id = 1;
  repeated SensorReading readings = 2;
}

message Ack {
  bool ok = 1;
}